shrine/lib/category_menu_page.dart

```diff
@@ -13,55 +13,53 @@
 // limitations under the License.
 
 import 'package:flutter/material.dart';
-import 'package:meta/meta.dart';
+import 'package:scoped_model/scoped_model.dart';
 
 import 'colors.dart';
+import 'model/app_state_model.dart';
 import 'model/product.dart';
 
 class CategoryMenuPage extends StatelessWidget {
-  final Category currentCategory;
-  final ValueChanged<Category> onCategoryTap;
   final List<Category> _categories = Category.values;
 
   const CategoryMenuPage({
     Key key,
-    @required this.currentCategory,
-    @required this.onCategoryTap,
-  })  : assert(currentCategory != null),
-        assert(onCategoryTap != null);
+  });
 
   Widget _buildCategory(Category category, BuildContext context) {
     final categoryString =
         category.toString().replaceAll('Category.', '').toUpperCase();
     final ThemeData theme = Theme.of(context);
-    return GestureDetector(
-      onTap: () => onCategoryTap(category),
-      child: category == currentCategory
-          ? Column(
-              children: <Widget>[
-                SizedBox(height: 16.0),
-                Text(
-                  categoryString,
-                  style: theme.textTheme.body2,
-                  textAlign: TextAlign.center,
-                ),
-                SizedBox(height: 14.0),
-                Container(
-                  width: 70.0,
-                  height: 2.0,
-                  color: kShrinePink400,
-                ),
-              ],
-            )
-          : Padding(
-              padding: EdgeInsets.symmetric(vertical: 16.0),
-              child: Text(
-                categoryString,
-                style: theme.textTheme.body2
-                    .copyWith(color: kShrineBrown900.withAlpha(153)),
-                textAlign: TextAlign.center,
-              ),
-            ),
+    return ScopedModelDescendant<AppStateModel>(
+      builder: (context, child, model) => GestureDetector(
+            onTap: () => model.setCategory(category),
+            child: model.selectedCategory == category
```

Comment:
[nit] One of these days, we should refactor this. Something like having a (private?) sub-widget, or something like `_buildCategory(Category category, bool selected)`.

Reply:
Eesh, yeah, that's a big builder. There's parts of this I'm reluctant to change because they're code from the codelab. Once the team is done with 105, though, we should have code parity between this version and the final codelab one, and can suggest changes from either side.